cilium-dbg/cmd/encrypt_status.go

```diff
@@ -223,17 +251,21 @@ func getDecryptionInterfaces() ([]string, error) {
 }
 
 func printEncryptionStatus(status models.EncryptionStatus) {
+	if status.Msg != "" {
+		fmt.Printf("Msg: %s\n", status.Msg)
+	}
+
 	fmt.Printf("Encryption: %-26s\n", status.Mode)
-	switch status.Mode {
```

Comment:
I'd suggest trying to use `status.Mode` from agent reply, or to set it after the inference, and preserve the switch here. That would avoid dumping info for both IPSec and Wireguard when one of them is not enabled (dirty data from previous state).

For instance, if the WireGuard inference did not fail (interface exists and wgClient.Device(types.IfaceName) do not fail) we set mode to Wireguard. Same for IPSec.

Reply:
i think displaying erratic data that was left over still has its value as it points out to a failure in the agent, which didn't clear up state when it was supposed to.
there's no inference on the encryption type, though. if the agent does not respond, the field is not populated. that way, we can both know what mode is explicitly enabled and whether there is any state on the not enabled feature. an operator could take that as a hint that there is a failure in the agent instead of silently ignoring the stale state (esp since this is a debug command, taken during times people want to understand what really is going on)

Reply:
It makes sense, it is just that we don't officially support switching encryption type in the same cluster 🤔
Anyway, feels like that's a good way to go to inspect such possible situation.
Feel free to solve this conversation.